pkg/apis/peakq/v0alpha1/types.go

```diff
@@ -0,0 +1,101 @@
+package v0alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/grafana/grafana-plugin-sdk-go/data"
+
+	common "github.com/grafana/grafana/pkg/apis/common/v0alpha1"
+)
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+type QueryTemplate struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec QueryTemplateSpec `json:"spec,omitempty"`
+}
+
+type QueryTemplateSpec struct {
+	Title string `json:"title"`
+
+	// The variables that can be used to render
+	// +listType=map
+	// +listMapKey=key
+	Variables []QueryVariable `json:"vars,omitempty"`
+
+	// Output variables
+	// +listType=set
+	Targets []Target `json:"targets"`
+}
+
+type Target struct {
+	// DataType is the returned Dataplane type from the query.
+	DataType data.FrameType `json:"dataType,omitempty"`
+
+	// DataTypeVersion is the version for the Dataplane type.
+	// TODO 2[uint] seems to panic, maybe implement DeepCopy on data.FrameTypeVersion?
+	// DataTypeVersion data.FrameTypeVersion `json:"dataTypeVersion,omitempty"`
+
+	Properties common.Unstructured `json:"properties"`
+}
+
+// QueryVariable is the definition of a variable that will be interpolated
+// in targets.
+type QueryVariable struct {
+	// Key is the name of the variable.
+	Key string `json:"key"`
+
+	// DefaultValue is the value to be used when there is no selected value
+	// during render.
+	DefaultValue string `json:"defaultValue"`
+
+	// Positions is a list of where to perform the interpolation
+	// within targets during render.
+	// The first string is the Idx of the target as a string, since openAPI
+	// does not support ints as map keys
+	// +listType=atomic
+	Positions map[string]map[Path][]Position `json:"positions"`
```

Comment:
I will be putting this back to where there targetIdx and Path are properties of position, and this is `Positions []Position`.

The map is something I need to build for render processing, it does not need to be part of the API (and this isn't quite the right way to do that map either - got tired an took a wrong turn :P